apps/src/templates/projects/SectionProjectsList.jsx

```diff
@@ -0,0 +1,105 @@
+import React from 'react';
+import ProjectsList from './ProjectsList';
+import _ from 'lodash';
+import color from "../../util/color";
+
+const ALL_STUDENTS = '_all_students';
+
+const styles = {
+  filterDropdown: {
+    margin: 0,
+    color: 'dimgray'
+  },
+  filterRow: {
+    backgroundColor: color.teal,
+    borderBottom: 'solid 1px white',
+    color: color.white,
+    padding: 10,
+    fontSize: 14
+  },
+  filterSpan: {
+    float: 'right',
+    display: 'inline-flex',
+    alignItems: 'center',
+  },
+  clearDiv: {
+    clear: 'both'
+  }
+};
+
+class SectionProjectsList extends React.Component {
+  constructor(props) {
+    super(props);
+
+    const studentNames = this.getStudentNames(props.projectsData);
+
+    this.state = {
+      studentNames,
+      selectedStudent: ALL_STUDENTS,
+    };
+  }
+
+  getStudentNames(projectsData) {
+    const studentsMap = {};
+    projectsData.forEach(project => {
+      studentsMap[project['studentName']] = true;
+    });
+    return Object.keys(studentsMap);
+  }
+
+  onChangeStudent(event) {
+    const selectedStudent = event.target.value;
+    this.setState({selectedStudent});
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (_.isEqual(nextProps, this.props)) {
+      return;
```

Comment:
This is not implicit to react. see here: https://facebook.github.io/react/docs/react-component.html#componentwillreceiveprops 

"React may call this method even if the props have not changed, so make sure to compare the current and next values if you only want to handle changes. "